ivadomed/loader/loader.py

```diff
@@ -208,10 +213,14 @@ def get_pair_data(self):
         if self.gt_handle is None:
             gt_data = None
         for gt in self.gt_handle:
-            if gt is not None:
+            if gt is not None and self.soft_input==False:
                 hwd_oriented = imed_loader_utils.orient_img_hwd(gt.get_fdata(cache_mode, dtype=np.float32),
                                                                 self.slice_axis)
                 gt_data.append(hwd_oriented.astype(np.uint8))
+            elif gt is not None and self.soft_input:
```

Comment:
instead of adding a `elif here` we could add a line like this: `data_type = np.float if self.soft_input else np.uint8`, `gt_data.append(hwd_oriented.astype(data_type))`. Currently this line is duplicated, `hwd_oriented = imed_loader_utils.orient_img_hwd(gt.get_fdata(cache_mode, dtype=np.float32),                                                             self.slice_axis)`, so this suggestion will prevent this duplication

Reply:
You are right ! Thanks !